skipruntime-ts/api/src/api.ts

```diff
@@ -152,31 +150,31 @@ export class NonUniqueValueException extends Error {}
 /**
  * A mutable iterator with at least one element
  */
-export interface NonEmptyIterator<T> extends Iterable<T> {
+export interface NonEmptyIterator<T> extends Iterable<T & Param> {
```

Comment:
I wasn't expecting this. Doesn't the `T` being a subtype of `Param` in the output iterable come from the type param of `NonEmptyIterator`? This makes it look like `NonEmptyIterator` internally deep-freezes the elements, rather than requiring that the type parameter is already a subtype of `Param`.